src/table/components/head/MenuList/RecursiveMenuList.tsx

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable import/no-cycle */
+import React from 'react';
+import Menu from '@mui/material/Menu';
+import { PopoverOrigin } from '@mui/material';
+import { MenuItemGroup } from '../../../types';
+import MenuGroupWrapper from './MenuGroupWrapper';
+
+interface RecursiveMenuListProps {
+  open: boolean;
+  anchorEl: HTMLDivElement | null;
+  onClose: () => void;
+  menuGroups: MenuItemGroup[];
+  transformOrigin?: PopoverOrigin; // eslint-disable-line react/require-default-props
+  anchorOrigin?: PopoverOrigin; // eslint-disable-line react/require-default-props
+  ariaLabel?: string; // eslint-disable-line react/require-default-props
+}
+
+const RecursiveMenuList = ({
+  anchorEl,
+  open,
+  onClose,
+  ariaLabel,
+  menuGroups,
+  transformOrigin,
+  anchorOrigin,
+}: RecursiveMenuListProps) => {
+  if (!menuGroups.length) return null;
+  return (
+    <Menu
+      className="sn-table-head-menu"
+      open={open}
+      anchorEl={anchorEl}
+      onClose={onClose}
+      autoFocus={false}
+      {...(ariaLabel ? { [`aria-labelledby`]: ariaLabel } : {})}
```

Comment:
Can you not do like this?

```suggestion
     ariaLabelledby={ariaLabel} // It's either undefined or a string
```